### PR TITLE
docs(clients): Add example code to deprecate DialSlash

### PR DESCRIPTION
**Pull request by bucanero**
Fixes DOC-241

### CODE REVIEW

**Review by anurags92 — APPROVED** [review]

File: content/clients/go.md [hunk 1]
```diff
@@ -92,6 +92,32 @@ Once logged in, the client can perform all the operations allowed to the `groot`
...
+	creds := credentials.NewClientTLSFromCert(pool, "")
+	return grpc.Dial(
+		host,
+		grpc.WithTransportCredentials(creds),
+		grpc.WithPerRPCCredentials(&authCreds{key}),
+	)
+}
+```
+
+
```
Comment (abhimanyusinghgaur) [inline content/clients/go.md:120]: No need to add this code, instead just add a note about deprecation.

File: content/clients/go.md [hunk 2]
```diff
@@ -92,6 +92,32 @@ Once logged in, the client can perform all the operations allowed to the `groot`
...
 ```go
 // This example uses dgo
 conn, err := dgo.DialSlashEndpoint("https://frozen-mango.eu-central-1.aws.cloud.dgraph.io/graphql", "<api-key>")
```
Comment (abhimanyusinghgaur) [inline content/clients/go.md:123]:
```suggestion
conn, err := dgo.DialCloud("https://frozen-mango.eu-central-1.aws.cloud.dgraph.io/graphql", "<api-key>")
```

File: content/clients/java.md [hunk 3]
```diff
@@ -84,6 +84,33 @@ Once logged-in, the `dgraphClient` object can be used to do any further operatio
...
+    Metadata metadata = new Metadata();
+    metadata.put(
+        Metadata.Key.of(gRPC_AUTHORIZATION_HEADER_NAME, Metadata.ASCII_STRING_MARSHALLER), apiKey);
+    return MetadataUtils.attachHeaders(
+        DgraphGrpc.newStub(
+            ManagedChannelBuilder.forAddress(gRPCAddress, 443).useTransportSecurity().build()),
+        metadata);
+  }
+```
+
```
Comment (abhimanyusinghgaur) [inline content/clients/java.md:113]: No need to add this, instead can add the following note about deprecation:
```
`DgraphClient.clientStubFromSlashEndpoint()` has been deprecated and will be removed in 21.07 release. Please use `DgraphClient.clientStubFromCloudEndpoint()` instead.
```
Same note can be formatted for other clients as well using the client-specific function name.

File: content/clients/java.md [hunk 4]
```diff
@@ -84,6 +84,33 @@ Once logged-in, the `dgraphClient` object can be used to do any further operatio
...
 ```java
 DgraphStub stub = DgraphClient.clientStubFromSlashEndpoint("https://civic-wine.us-west-2.aws.cloud.dgraph.io/graphql", "your-api-key");
```
Comment (abhimanyusinghgaur) [inline content/clients/java.md:115]:
```suggestion
DgraphStub stub = DgraphClient.clientStubFromCloudEndpoint("https://civic-wine.us-west-2.aws.cloud.dgraph.io/graphql", "your-api-key");
```

File: content/clients/javascript/grpc.md [hunk 5]
```diff
@@ -81,6 +81,35 @@ Once logged in, the client can perform all the operations allowed to the `groot`
...
+        },
+    );
+    const credentials = grpc.credentials.combineChannelCredentials(
+        grpc.credentials.createSsl(),
+        metaCreds,
+    );
+    return new DgraphClientStub(backenedURL, credentials);
+}
+```
+
```
Comment (abhimanyusinghgaur) [inline content/clients/javascript/grpc.md:112]: No need to add this part, just add the deprecation note. The JavaScript client doesn't yet seem to have the new method, so just skip the new method name in the deprecation note.

File: content/clients/python.md [hunk 6]
```diff
@@ -62,6 +62,22 @@ Once logged in, the client can perform all the operations allowed to the `groot`
...
+    creds = grpc.ssl_channel_credentials()
+    call_credentials = grpc.metadata_call_credentials(
+        lambda context, callback: callback((("authorization", api_key),), None))
+    composite_credentials = grpc.composite_channel_credentials(
+        creds, call_credentials)
+    client_stub = DgraphClientStub('{host}:{port}'.format(
+        host=host, port="443"), composite_credentials, options=(('grpc.enable_http_proxy', 0),))
+    return client_stub
+```
+
```
Comment (abhimanyusinghgaur) [inline content/clients/python.md:80]: No need to add this, just add the deprecation note.

File: content/clients/python.md [hunk 7]
```diff
@@ -62,6 +62,22 @@ Once logged in, the client can perform all the operations allowed to the `groot`
...
 ```python
 import pydgraph
```
Comment (abhimanyusinghgaur) [inline content/clients/python.md:83]:
```suggestion
client_stub = pydgraph.DgraphClientStub.from_cloud("https://frozen-mango.eu-central-1.aws.cloud.dgraph.io/graphql", "<api-key>")
```

This was line 84, GitHub is weird, ain't letting me suggest on that line.

**Review by abhimanyusinghgaur — COMMENTED** [review]
I think, we deprecated `DialSlashEndpoint`, and renamed it to `DialCloud` (notice `Slash` -> `Cloud` renaming). So, it doesn't make sense to add the code snippets of the old functions for every language. Rather, we should be changing the usage of the function for every language from the old to the new name in the docs and change `Slash` to `Cloud` wherever it is used in terminology.
Here's the related PRs for every client:
* go: https://github.com/dgraph-io/dgo/pull/147/files
* java: https://github.com/dgraph-io/dgraph4j/pull/170/files
* javascript: I didn't find the new function, but the old function has been deprecated as per this: https://github.com/dgraph-io/dgraph-js/pull/136
* python: https://github.com/dgraph-io/pydgraph/pull/169/files

Now, would it be a good idea to add a note in every language docs that the old function has been deprecated, please use the new API instead? I am adding a sample comment for the Java client, maybe the same can be added for the other client docs too.

**Pull request merged**